misk/src/main/kotlin/misk/jdbc/DataSourceConfig.kt

```diff
@@ -6,18 +6,26 @@ import java.time.Duration
 /** Defines a type of datasource */
 enum class DataSourceType(
   val driverClassName: String,
+  val hibernateDialect: String,
   val buildJdbcUrl: (DataSourceConfig) -> String
 ) {
-  MYSQL("com.mysql.jdbc.Driver", { config ->
-    val port = config.port ?: 3306
-    val host = config.host ?: "127.0.0.1"
-    val database = config.database ?: ""
-    "jdbc:mysql://${host}:$port/$database"
-
-  }),
-  HSQLDB("org.hsqldb.jdbcDriver", { config ->
-    "jdbc:hsqldb:mem:${config.database!!}"
-  })
+  MYSQL(
+      driverClassName = "com.mysql.jdbc.Driver",
+      hibernateDialect = "org.hibernate.dialect.MySQL57Dialect",
+      buildJdbcUrl = { config ->
+        val port = config.port ?: 3306
+        val host = config.host ?: "127.0.0.1"
+        val database = config.database ?: ""
+        "jdbc:mysql://$host:$port/$database"
+      }
+  ),
+  HSQLDB(
+      driverClassName = "org.hsqldb.jdbcDriver",
+      hibernateDialect = "org.hibernate.dialect.H2Dialect",
+      buildJdbcUrl = { config ->
+        "jdbc:hsqldb:mem:${config.database!!}"
```

Comment:
We might want to set ``sql.syntax_mys=true`` on the JDBC URL to get MySQL compatible syntax (``[TINY|SMALL|MEDIUM]BLOB``, ``AUTO_INCREMENT``, etc).

Reply:
Nice.

Reply:
Fixed.